diff --git a/giscanner/msvccompiler.py b/giscanner/msvccompiler.py
index c9f14b58..1c4bd9b3 100644
--- a/giscanner/msvccompiler.py
+++ b/giscanner/msvccompiler.py
@@ -40,7 +40,13 @@ class MSVCCompiler(DistutilsMSVCCompiler):
 
     def __init__(self, verbose=0, dry_run=0, force=0):
         super(DistutilsMSVCCompiler, self).__init__()
-        CCompiler.__init__(self, verbose, dry_run, force)
+        try:
+            CCompiler.__init__(self, verbose, dry_run, force)
+        except TypeError:
+            CCompiler.__init__(self,
+                               verbose=verbose,
+                               dry_run=dry_run,
+                               force=force)
         self.__paths = []
         self.__arch = None  # deprecated name
         self.initialized = False
