diff --git a/gi/__init__.py b/gi/__init__.py
index 826489a1..afba7884 100644
--- a/gi/__init__.py
+++ b/gi/__init__.py
@@ -23,6 +23,7 @@ from pkgutil import extend_path
 __path__ = extend_path(__path__, __name__)
 
 import sys
+import sysconfig
 import os
 import importlib
 import types
@@ -36,8 +37,23 @@ _static_binding_error = ('When using gi.repository you must not import static '
 if 'gobject' in sys.modules:
     raise ImportError(_static_binding_error)
 
+# Windows official Python requires gi DLLs to be loaded manually
+if sysconfig.get_platform().startswith("win"):
+    env_path = os.environ.get("PATH", "").split(os.pathsep)
+    first_gtk_dll_path = next(
+        filter(
+            lambda path: path is not None and os.path.isfile(os.path.join(path, "girepository-1.0-1.dll")),
+            env_path,
+        ),
+        None,
+    )
+    if first_gtk_dll_path:
+        with os.add_dll_directory(first_gtk_dll_path):
+            from . import _gi
+
+else:
+    from . import _gi
 
-from . import _gi
 from ._gi import _API  # noqa: F401
 from ._gi import Repository
 from ._gi import PyGIDeprecationWarning  # noqa: F401
